test/rekt/features/channel/data_plane.go

```diff
@@ -154,6 +157,115 @@ func channelAcceptsCEVersions(ctx context.Context, t feature.T) {
 	knconf.AcceptsCEVersions(ctx, t, channel_impl.GVR(), name)
 }
 
+func inputMethodNotPost(ctx context.Context, t feature.T) {
+	channelName := state.GetStringOrFail(ctx, t, ChannelableNameKey)
+
+	structuredcontenttype := "application/cloudevents+json"
+	bodycontent := `{
+    "specversion" : "1.0",
+    "type" : "sometype",
+    "source" : "json.request.sender.test.knative.dev",
+    "id" : "2222-4444-6666",
+    "time" : "2020-07-06T09:23:12Z",
+    "datacontenttype" : "application/json",
+    "data" : {
+        "message" : "helloworld"
+    }
+}`
+
+	source := feature.MakeRandomK8sName("source")
+
+	eventshub.Install(source,
+		eventshub.StartSenderToResource(channel_impl.GVR(), channelName),
+		eventshub.InputHeader("content-type", structuredcontenttype),
+		eventshub.InputBody(bodycontent),
+		eventshub.InputMethod("GET"),
+	)(ctx, t)
+
+	store := eventshub.StoreFromContext(ctx, source)
+	events := knconf.Correlate(store.AssertAtLeast(t, 2, knconf.SentEventMatcher("")))
+	for _, e := range events {
+		if e.Response.StatusCode == 405 {
```

Comment:
```suggestion
		if e.Response.StatusCode != 405 {
```
@Vishal-Chdhry am I misunderstanding something here or should this be a not equal check?